cdap-app-templates/cdap-etl/cdap-etl-proto/src/main/java/co/cask/cdap/etl/proto/Connection.java

```diff
@@ -28,15 +28,21 @@
   private final String from;
   private final String to;
   private final String port;
+  private final Boolean condition;
 
   public Connection(String from, String to) {
     this(from, to, null);
   }
 
   public Connection(String from, String to, @Nullable String port) {
+    this(from, to, port, null);
+  }
+
+  public Connection(String from, String to, @Nullable String port, @Nullable Boolean condition) {
```

Comment:
a connection should not have a non-null port and a non-null condition. Instead, we can have this as private and a public one with from, to, and condition.